open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/virtualization-solution-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/virtulaizationsolutionconnector/derby/DerbyConnector.java

```diff
@@ -0,0 +1,81 @@
+package org.odpi.openmetadata.openconnectors.governancedaemonconnectors.viewgenerator.virtulaizationsolutionconnector.derby;
+
+import org.odpi.openmetadata.frameworks.connectors.ConnectorBase;
+import org.odpi.openmetadata.frameworks.connectors.properties.AdditionalProperties;
+import org.odpi.openmetadata.frameworks.connectors.properties.ConnectionProperties;
+import org.odpi.openmetadata.frameworks.connectors.properties.EndpointProperties;
+import org.odpi.openmetadata.openconnectors.governancedaemonconnectors.viewgenerator.virtulaizationsolutionconnector.VirtualizationSolutionConnectorBase;
+import org.odpi.openmetadata.openconnectors.governancedaemonconnectors.viewgenerator.virtulaizationsolutionconnector.model.LogicTable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DerbyConnector extends ConnectorBase implements VirtualizationSolutionConnectorBase{
+
+    /*
+     * Variables fpr building the connection to the Derby database
+     */
+    private String serverAddress;
+    private String username;
+    private String password;
+    private int    timeoutInSecond;
+    private String logicTableName;
+    private String logicTableDefinition;
+
+    private static final Logger log = LoggerFactory.getLogger(DerbyConnector.class);
+
+
+    /**
+     * Initialize the connector.
+     *
+     * @param connectorInstanceId - unique id for the connector instance - useful for messages etc
+     * @param connectionProperties - POJO for the configuration used to create the connector.
+     */
+    @Override
+    public void initialize(String connectorInstanceId, ConnectionProperties connectionProperties){
+        super.initialize(connectorInstanceId, connectionProperties);
+
+        AdditionalProperties additionalProperties = connectionProperties.getAdditionalProperties();
+        EndpointProperties   endpoint             = connectionProperties.getEndpoint();
+
+        if (endpoint != null){
+            serverAddress = endpoint.getAddress();
+            AdditionalProperties serverProperty = endpoint.getAdditionalProperties();
+            if (serverProperty != null){
+                username = serverProperty.getProperty("username").toString();
```

Comment:
The connection object has userId and password in it - no need to use the endpoint additional properties (endpoints can be shared between connections)

Reply:
Hi Mandy, I tried to store the username and password in the connection object but the getConnectionBean  is set as [Protected](https://github.com/odpi/egeria/tree/master/open-metadata-implementation/frameworks/open-connector-framework/src/main/java/org/odpi/openmetadata/frameworks/connectors/properties/ConnectionProperties.java). I couldn't load the userId and password for the Connector. Could you please advise how to solve the problem? Thanks!

Reply:
Where are you calling getConnectionBean from?
The API for the connection object has these fields public:
```
 /**
     * Return id of the calling user.
     *
     * @return string
     */
    public String getUserId()
    {
        return userId;
    }

    /**
     * Set up the id of the calling user.
     *
     * @param userId string
     */
    public void setUserId(String userId)
    {
        this.userId = userId;
    }

    /**
     * Return an encrypted password.  The caller is responsible for decrypting it.
     *
     * @return string
     */
    public String getEncryptedPassword()
    {
        return encryptedPassword;
    }

    /**
     * Set up an encrypted password.
     *
     * @param encryptedPassword string
     */
    public void setEncryptedPassword(String encryptedPassword)
    {
        this.encryptedPassword = encryptedPassword;
    }
```
These values can then be set in the connection by the caller that is creating the Derby connector through ConnectorBroker.getConnector().  The connector itself can extract these values using the methods on the connectionProperties passed on the initialize() request.

For example:
```
public void initialize(String               connectorInstanceId,
                           ConnectionProperties connectionProperties)
    {
        this.connectorInstanceId = connectorInstanceId;
        this.connectionProperties = connectionProperties;
        
        String userId = connectionProperties.getUserId();
```

Reply:
Hi Mandy, regarding the https://github.com/odpi/egeria/blob/master/open-metadata-implementation/frameworks/open-connector-framework/src/main/java/org/odpi/openmetadata/frameworks/connectors/properties/ConnectionProperties.java, the method to get the Connection object is protected. Method getUserId() is located in the Class Connection. So I hope to use 

Connection connection = connectionProperties.getConnectionBean();
String userId = connection.getUserId();

But getConnectionBean() is protected.

Reply:
You should be using the connectionProperties not the getConnectionBean().  The getUserId() method on connectionProperties delegates to the connection bean.

Reply:
Hi Mandy, regarding the https://github.com/odpi/egeria/blob/master/open-metadata-implementation/frameworks/open-connector-framework/src/main/java/org/odpi/openmetadata/frameworks/connectors/properties/ConnectionProperties.java, there is no method getUserId... May I know if you are referring to another class? Or do I need to add corresponding public methods to fetch user ID and password? Thanks and have a nice day!

Reply:
Sorry, my mistake, these methods are in my fork but I have not pushed them to master.  That is why they are workign for me and not you :) will get them into master ASAP

Reply:
No worries. :) After you merge your fork, I will update the code.